src/libxively/sft/xi_sft_logic.c

```diff
@@ -131,6 +144,89 @@ xi_sft_send_file_status( const xi_sft_context_t* context,
     ( *context->fn_send_message )( context->send_message_user_data, message_file_status );
 }
 
+xi_state_t xi_sft_select_next_resource_to_download( xi_sft_context_t* context )
```

Comment:
could you please add unit tests on this function. In the unit test case setups you could add any weird index map: out of bounds, readding the same index again, and/or leaving some indices out replacing them with redundant indices

Reply:
also I've started to add `_` (underscore) prefixes for .c private functions, also a `static` classifier. This `static` keyword makes the function not visible from outside of the compilation / translation unit. But yes, if you make this function static, then the unit test won't be able to re-use it in the test cases. So this `static` stuff does not play here but the `_` prefix still can stay in the game.